examples/ewf/molecules/25-externally-correct.py

```diff
@@ -0,0 +1,62 @@
+import numpy as np
+import pyscf
+import pyscf.gto
+import pyscf.scf
+import pyscf.cc
+import pyscf.fci
+import vayesta
+import vayesta.ewf
+from vayesta.misc import molecules
+
+mol = pyscf.gto.Mole()
+mol.atom = """
+Se	0.0000	0.0000	0.2807
+O 	0.0000	1.3464	-0.5965
+O 	0.0000	-1.3464	-0.5965
+"""
+mol.basis = 'cc-pVDZ'
+mol.output = 'pyscf.out'
+mol.build()
+
+# Hartree-Fock
+mf = pyscf.scf.RHF(mol)
+mf.kernel()
+
+# Reference full system CCSD:
+cc = pyscf.cc.CCSD(mf)
+cc.kernel()
+
+# 1) Consider setup where you have a complete CCSD, externally corrected by local atomic fragment+DMET FCI clusters
+emb = vayesta.ewf.EWF(mf)
+with emb.iao_fragmentation() as f:
+    # Add all atomic FCI fragments with DMET bath
+    # Store the FCI wave functions as CCSDTQ types, so they can be used for correction later.
+    # These want to be flagged as 'auxiliary', as they are solved first, and then used as constraints for the 
+    # non-auxiliary fragments.
+    fci_frags = f.add_all_atomic_fragments(solver='FCI', bath_options=dict(bathtype='dmet'), store_wf_type='CCSDTQ', auxiliary=True)
+    # Add single 'complete' CCSD fragment covering all IAOs
+    ccsd_frag = f.add_full_system(solver='CCSD', bath_options=dict(bathtype='full'))
+# Setup the external correction from the CCSD fragment.
+# Two main options: correction_type='external-ccsdv' and 'external-fciv'. For the important T3 * V contribution to the
+#   T2 amplitudes, this determines whether the V is expressed in the FCI or CCSD cluster space. The CCSD cluster is
+#   larger, and hence this is likely to be better, as the correction is longer-ranged (though slightly more expensive).
+#   The other option is 'projectors', which should be an integer between 0 and 2 (inclusive).
+#   The larger the number, the more fragment projectors are applied to the correcting T2 contributions, and less
+#   'bath' correlation from the FCI clusters is used as a constraint in the external correction of the CCSD clusters.
+#   For multiple constraining fragments, proj=0 will double-count the correction due to overlapping bath spaces, and 
+#   in this case, only proj=1 will be exact in the limit of enlarging (FCI) bath spaces.
+ccsd_frag.add_external_corrections(fci_frags, correction_type='external-ccsdv', projectors=1)
```

Comment:
Are 'external-ccsdv' and 'external-fciv' good names for this option? - basically, I am asking if it is conceivable to have an external correction in this spirit with a different low-level method than CCSD or a different high-level method than FCI - in this case, and in order to future proof keyword arguments, it would be better to refer to 'low_level' or 'high_level' (or similar names), rather than to CCSD and FCI directly.

Reply:
In the latest commit, I've removed the different options of 'external-ccsdv' and 'external-fciv', so there is just the 'external' option as an external correction. The choice of 'ccsdv' and 'fciv' is controlled by an additional optional argument to the add_external_corrections function, `low_level_coul`, where True is equivalent to 'external-ccsdv' and False 'external-fciv'. Should be clearer. Tests and examples updated.